### PR TITLE
Cosmos CFP: fix for stale lease and remove blocking call

### PR DESCRIPTION
**Pull request by milismsft**
Fix for CFP host failing to take over stale leases that still have ownership set.
Remove last blocking call (see #5013)

### CODE REVIEW

File: sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/changefeed/implementation/LeaseStoreManagerImpl.java [hunk 1]
```diff
@@ -435,16 +435,15 @@ public LeaseStoreManagerBuilderDefinition hostName(String hostName) {
...
 
     private String getDocumentId(String leaseToken)
     {
         return this.getPartitionLeasePrefix() + leaseToken;
     }
 
-    private String getPartitionLeasePrefix()
-    {
+    private String getPartitionLeasePrefix() {
         return this.settings.getContainerNamePrefix() + "..";
```
Comment (moderakh) [inline sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/changefeed/implementation/LeaseStoreManagerImpl.java]: we should consider making these as defined constants.

File: sdk/cosmos/microsoft-azure-cosmos/src/test/java/com/azure/data/cosmos/rx/ChangeFeedProcessorTest.java [hunk 2]
```diff
@@ -40,7 +50,7 @@
     private CosmosContainer createdFeedCollection;
     private CosmosContainer createdLeaseCollection;
     private List<CosmosItemProperties> createdDocuments;
-    private static Map<String, CosmosItemProperties> receivedDocuments;
+    private static volatile Map<String, CosmosItemProperties> receivedDocuments;
```
Comment (moderakh) [inline sdk/cosmos/microsoft-azure-cosmos/src/test/java/com/azure/data/cosmos/rx/ChangeFeedProcessorTest.java]: why is this volatile?

Comment (milismsft) [reply sdk/cosmos/microsoft-azure-cosmos/src/test/java/com/azure/data/cosmos/rx/ChangeFeedProcessorTest.java]: The initialization of the map was happening inside a Reactor thread; this is no longer the case, so I'll remove the "volatile".

File: sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/changefeed/implementation/LeaseStoreManagerImpl.java (hunk 1 above)
Comment (milismsft) [reply sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/changefeed/implementation/LeaseStoreManagerImpl.java]: changed to a constant

**Pull request merged**